### PR TITLE
Fixed Salesforce Source not running sync

### PR DESCRIPTION
**Pull request by Luishfs**
# Description

Fixed Salesforce Source not runing FULL_TABLE/INCREMENTAL sync
Updated README

# How Has This Been Tested?

Tested using Salesforce Source -> PostgreSQL Destination

## FULL TABLE sync
Screenshot from 2023-11-21 21-56-10

Screenshot from 2023-11-21 21-56-30

## INCREMENTAL sync
No bookmarks were selected since the Salesforce Source defines and uses internal bookmarks as default

Screenshot from 2023-11-21 22-02-48

Screenshot from 2023-11-21 22-03-24

# Checklist
- [X] The PR is tagged with proper labels (bug, enhancement, feature, documentation)
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] If new documentation has been added, relative paths have been added to the appropriate section of `docs/mint.json`

cc:

@wangxiaoyou1993

### CODE REVIEW

File: mage_integrations/mage_integrations/sources/salesforce/README.md [hunk 1]
```diff
@@ -11,13 +11,35 @@ You must enter the following settings when configuring this source:
 | Key | Description | Sample value
 | --- | --- | --- |
 | `api_type` | The `api_type` is used to switch the behavior of the tap between using Salesforce's "REST" and "BULK" APIs. When new fields are discovered in Salesforce objects, the `select_fields_by_default` key describes whether or not the tap will select those fields by default. | `REST`, `BULK` |
-| `client_id` | OAuth Salesforce App secrets. | `ABC1...` |
-| `client_secret` | OAuth Salesforce App secrets. | `ABC1...` |
-| `is_sandbox` | If `true`, then the sandbox account will be used to load dat from. | `true`, `false` |
-| `lookback_window` | The `lookback_window` (in seconds) subtracts the desired amount of seconds from the bookmark to sync past data. Recommended value: 10 seconds. | `10` |
-| `refresh_token` | The `refresh_token` is a secret created during the OAuth flow | `ABC1...` |
+| `credentials` | Dict containing simple-salesforce login data, can be either `OAUTH` or `PASSWORD`, please check below for detailed info.
```
Comment (wangxiaoyou1993) [inline mage_integrations/mage_integrations/sources/salesforce/README.md:14]: Let's make the config backward compatible.

File: mage_integrations/mage_integrations/sources/salesforce/client/tap_salesforce/sync.py [hunk 2]
```diff
@@ -53,14 +44,9 @@ def get_stream_version(catalog_entry, state):
...
 
-
 def resume_syncing_bulk_query(sf, catalog_entry, job_id, state, counter):
     bulk = Bulk(sf)
-    current_bookmark = singer.get_bookmark(state, catalog_entry['tap_stream_id'],
-                                           'JobHighestBookmarkSeen') or sf.get_start_date(
-                                                                                    state,
-                                                                                    catalog_entry
-                                                                                         )
+    current_bookmark = singer.get_bookmark(state, catalog_entry['tap_stream_id'], 'JobHighestBookmarkSeen') or sf.get_start_date(state, catalog_entry)
```
Comment (wangxiaoyou1993) [inline mage_integrations/mage_integrations/sources/salesforce/client/tap_salesforce/sync.py:49]: this line is too long. can you lint the files with flake8?

File: mage_integrations/mage_integrations/sources/salesforce/client/tap_salesforce/__init__.py [hunk 3]
```diff
@@ -1,21 +1,27 @@
 #!/usr/bin/env python3
+from __future__ import annotations
```
Comment (wangxiaoyou1993) [inline mage_integrations/mage_integrations/sources/salesforce/client/tap_salesforce/__init__.py:2]: where is it used?

File: mage_integrations/mage_integrations/sources/salesforce/client/tap_salesforce/__init__.py [hunk 4]
```diff
@@ -1,21 +1,27 @@
...
-from typing import List
+from copy import deepcopy
 
 import singer
 import singer.utils as singer_utils
 from singer import metadata, metrics
 
-from mage_integrations.sources.messages import write_schema
-from mage_integrations.sources.salesforce.client.tap_salesforce import salesforce
+import mage_integrations.sources.salesforce.client.tap_salesforce.salesforce as salespath
```
Comment (wangxiaoyou1993) [inline mage_integrations/mage_integrations/sources/salesforce/client/tap_salesforce/__init__.py:14]: why do you need to rename it?

**Review by wangxiaoyou1993 — COMMENTED** [review]
Can you describe what was the issue and which code fixes it?
I don't know what changes I should pay more attention to during code review.

File: mage_integrations/mage_integrations/sources/salesforce/README.md (hunk 1 above)
Comment (Luishfs) [reply mage_integrations/mage_integrations/sources/salesforce/README.md]: Added in 8cc2a0e2931ffc6e60ebd0611b0fa1521525ccda

File: mage_integrations/mage_integrations/sources/salesforce/client/tap_salesforce/sync.py (hunk 2 above)
Comment (Luishfs) [reply mage_integrations/mage_integrations/sources/salesforce/client/tap_salesforce/sync.py]: Linted in 8cc2a0e2931ffc6e60ebd0611b0fa1521525ccda

File: mage_integrations/mage_integrations/sources/salesforce/client/tap_salesforce/__init__.py (hunk 3 above)
Comment (Luishfs) [reply mage_integrations/mage_integrations/sources/salesforce/client/tap_salesforce/__init__.py]: This was added from the tap-salesforce source repo, removed in 8cc2a0e2931ffc6e60ebd0611b0fa1521525ccda

File: mage_integrations/mage_integrations/sources/salesforce/client/tap_salesforce/__init__.py (hunk 4 above)
Comment (Luishfs) [reply mage_integrations/mage_integrations/sources/salesforce/client/tap_salesforce/__init__.py]: Removed in 8cc2a0e2931ffc6e60ebd0611b0fa1521525ccda

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**